File: client/common/graphql/mutations/userLogin.ts
import gql from 'graphql-tag';
import { USER_AUTH_FRAGMENT } from '../fragments';

const USER_LOGIN_MUTATION = gql`
	mutation userLogin(
		$email: String!
		$password: String!
	){
		userLogin(
			authInput:{
				email: $email, 
				password: $password
			}
		){
			...UserAuthDetails
		}
	}
	${USER_AUTH_FRAGMENT}
`;

export default USER_LOGIN_MUTATION;

File: app/graphql/mutations/user_login.rb
module Mutations
	class UserLogin < Mutations::BaseMutation
		argument :auth_input, Types::AuthEmailInput, required: true

		type Types::UserType

		def resolve(auth_input:)
			input = hash_to_camel_case(auth_input)
			@user = User::TokenAuth.find_for_authentication(email: input[:email])
			return GraphQL::ExecutionError.new('Incorrect Email or Password') unless @user && @user.authenticate(input[:password])
			set_auth_cookies
			return @user
		end

	end
end

